### PR TITLE
add textbox

### PR DESCRIPTION
**Pull request by BlackAnge**

### CODE REVIEW

File: app/views/tasks/_task.html.erb [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
-		<%= label_tag "checked_#{task.id}", "#{task.title}", class: 'completed-action title'  %>
+<% tasks.each do |task| %>
+	<div data-task-id='<%= task.id %>' class="row pad-top-8 task row-task <%= task.status %>">
+		<div class="col-md-9">
+			<%= check_box_tag 'status', value = task.status, checked = task.completed?, options = { id: "checked_#{task.id}", class: "checkbox-status-#{task.status}" } %>
+			<%= label_tag "checked_#{task.id}", "#{task.title}", class: 'completed-action title' %>
+		</div>
+		<div class="col-md-3">
+  		<%= link_to "Edit", edit_task_path(task), class: 'action icon-edit' %>
+  		<span class="delete-action action">Delete</span>
```
Comment (0bman) [inline app/views/tasks/_task.html.erb]: Use `link_to` method with option `remote: true`

Comment (0bman) [reply app/views/tasks/_task.html.erb]: And delete `remove ` method from [js file](https://github.com/BlackAnge/todo-list/pull/4/files#diff-4fa61dd3503d0b5c7141293f84f8dfa1R33)

File: app/assets/javascripts/tasks.js [hunk 2]
```diff
@@ -1,20 +1,63 @@
 function Task() {};
+var task = new Task();
+
+Task.prototype.init = function() {
+  task.active();
+  task.completed();
+  task.remove();
```
Comment (0bman) [inline app/assets/javascripts/tasks.js]: When you add `link_to` helper method in the partial after that you have to delete this line

File: app/assets/javascripts/tasks.js [hunk 3]
```diff
@@ -1,20 +1,63 @@
...
+      });
+    };
+  });
+}
+
+Task.prototype.submitDate = function(){
+  $(function() {
+    $('.datepicker').datepicker({
+      format: 'dd/mm/yyyy',
+      autoclose: true,
```
Comment (0bman) [inline app/assets/javascripts/tasks.js]: remove this option

File: app/assets/javascripts/tasks.js [hunk 4]
```diff
@@ -1,20 +1,63 @@
...
+    }).on('changeDate', function(event) {
+      $(this).focus();
+      if(event.which == 13) {
+        $('form').submit();
+      }
+      $('#input-text').focus();
```
Comment (0bman) [inline app/assets/javascripts/tasks.js]: move this line to condition

File: app/assets/javascripts/tasks.js [hunk 5]
```diff
@@ -1,20 +1,63 @@
...
+}
+
+Task.prototype.submitDate = function(){
+  $(function() {
+    $('.datepicker').datepicker({
+      format: 'dd/mm/yyyy',
+      autoclose: true,
+    }).on('changeDate', function(event) {
+      $(this).focus();
+      if(event.which == 13) {
```
Comment (0bman) [inline app/assets/javascripts/tasks.js]: space after `if`

File: app/assets/stylesheets/tasks.scss [hunk 6]
```diff
@@ -62,12 +62,38 @@ body {
     transition: 0.3s;
   }
 
+  form{
```
Comment (0bman) [inline app/assets/stylesheets/tasks.scss]: space

File: app/assets/stylesheets/tasks.scss [hunk 7]
```diff
@@ -62,12 +62,38 @@ body {
...
+
+  input[type=checkbox] {
+    display: none;
+  }
+  input[type=submit] {
+    visibility: hidden;
+    display: none;
+  }
+
+  .input-text{
```
Comment (0bman) [inline app/assets/stylesheets/tasks.scss]: spacee

File: app/assets/stylesheets/tasks.scss [hunk 8]
```diff
@@ -62,12 +62,38 @@ body {
...
+    background-image: url(arrow_down.svg);
+    background-repeat: no-repeat;
+    height: 32px;
+    padding-left: 58px;
+  }
+
+  .input-data{
```
Comment (0bman) [inline app/assets/stylesheets/tasks.scss]: space

File: app/models/task.rb [hunk 9]
```diff
@@ -1,9 +1,9 @@
 class Task < ApplicationRecord
   enum status: %i[active completed]
 
-  validates :title, :description, presence: true
-  validates :completed_to, presence: true
-  validate :future_completed_date
+  validates :title, presence: true
+  # validates :completed_to, presence: true
```
Comment (0bman) [inline app/models/task.rb]: uncomment it

File: app/models/task.rb [hunk 10]
```diff
@@ -1,9 +1,9 @@
...
+  # validate :future_completed_date
```
Comment (0bman) [inline app/models/task.rb]: and it too

File: app/views/tasks/_task.html.erb [hunk 11]
```diff
@@ -1,10 +1,12 @@
-<div data-task-id='<%= task.id %>' class="row pad-top-8 task row-task <%= task.status %>">
-	<div class="col-md-10">
-		<%= check_box_tag 'status', value = task.status, checked = task.completed?, options = { id: "checked_#{task.id}", class: "checkbox-status-#{task.status}" } %>
-		<%= label_tag "checked_#{task.id}", "#{task.title}", class: 'completed-action title'  %>
+<% tasks.each do |task| %>
+	<div data-task-id='<%= task.id %>' class="row pad-top-8 task row-task <%= task.status %>">
```
Comment (0bman) [inline app/views/tasks/_task.html.erb]: single quotes

File: app/views/tasks/_task.html.erb [hunk 12]
```diff
@@ -1,10 +1,12 @@
...
+		<div class="col-md-9">
```
Comment (0bman) [inline app/views/tasks/_task.html.erb]: single quotes

File: app/views/tasks/_task.html.erb [hunk 13]
```diff
@@ -1,10 +1,12 @@
...
+			<%= check_box_tag 'status', value = task.status, checked = task.completed?, options = { id: "checked_#{task.id}", class: "checkbox-status-#{task.status}" } %>
+			<%= label_tag "checked_#{task.id}", "#{task.title}", class: 'completed-action title' %>
+		</div>
+		<div class="col-md-3">
```
Comment (0bman) [inline app/views/tasks/_task.html.erb]: quotes

File: app/views/tasks/create.js.erb [hunk 14]
```diff
@@ -0,0 +1,3 @@
+$("#show").html("<%= escape_javascript(render partial: 'task', locals: { tasks: @tasks } ) %>");
```
Comment (0bman) [inline app/views/tasks/create.js.erb]: quotes

File: app/views/tasks/index.html.erb [hunk 15]
```diff
@@ -3,18 +3,24 @@
...
   </div>
 </div>
-
 <div class="row content-task">
   <% if @tasks.empty? %>
     <span class="text-warning">There are no tasks!</span>
   <% else %>
     <div class="row col-md-12 row-text">
-      <input type="text" name="massage" placeholder="What needs to be done?" class="col-md-12 border-0">
+      <%= form_for Task.new, method: :post, remote: true do |f| %>
```
Comment (0bman) [inline app/views/tasks/index.html.erb]: move it to variable

File: app/views/tasks/index.html.erb [hunk 16]
```diff
@@ -3,18 +3,24 @@
...
+        <div class="col-md-9 border-0">
```
Comment (0bman) [inline app/views/tasks/index.html.erb]: quotes

File: app/controllers/tasks_controller.rb [hunk 17]
```diff
@@ -1,18 +1,14 @@
...
     @task = Task.new
   end
 
   def create
     @task = Task.create(task_params)
-    if @task.errors.empty?
-      redirect_to root_path
-    else
-      render 'new'
+    @tasks = Task.all.order('created_at DESC')
```
Comment (0bman) [inline app/controllers/tasks_controller.rb]: Remove this line, it doesn't need

File: app/assets/javascripts/tasks.js [hunk 18]
```diff
@@ -1,20 +1,91 @@
...
+        $('form').submit();
+        $('#input-text').focus();
+      } else {
+        $input.focus();
+      };
+    })
+  })
+};
+
+Task.prototype.create = function(task_id,task_title, task_status, task_url) {
```
Comment (0bman) [inline app/assets/javascripts/tasks.js]: space after `task_id`

File: app/assets/javascripts/tasks.js [hunk 19]
```diff
@@ -1,20 +1,91 @@
...
+  var row = document.createElement("div");
```
Comment (0bman) [inline app/assets/javascripts/tasks.js]: single quotes

File: app/assets/javascripts/tasks.js [hunk 20]
```diff
@@ -1,20 +1,91 @@
...
+  var linkDelete = document.createElement("a");
+
+  row.setAttribute("data-task-id", task_id);
+  row.className = "row pad-top-8 task row-task " + task_status;
+  div1.className = "col-md-9";
+  div2.className = "col-md-3";
+  checkBox.type = "checkbox";
+  checkBox.id = ("checked_" + task_id);
+  checkBox.name = "status";
+  checkBox.value = "<%= @task.status %>";
```
Comment (0bman) [inline app/assets/javascripts/tasks.js]: wtf? `<%= @task.status %>` it's rails style

File: app/views/tasks/create.js.erb [hunk 21]
```diff
@@ -0,0 +1,6 @@
+$("#input-text").val("");
+$("#input-date").val("");
+
+var task = new Task();
+
+task.create("<%= @task.id %>", "<%= @task.title %>", "<%= @task.status %>", "<%= task_path(@task) %>");
```
Comment (0bman) [inline app/views/tasks/create.js.erb]: can we send params without double quotes?
`task.create(<%= @task.id %>, <%= @task.title %>, <%= @task.status %>, <%= task_path(@task) %>);`

File: app/views/tasks/index.html.erb [hunk 22]
```diff
@@ -3,18 +3,24 @@
      <h1>todos</h1>
   </div>
 </div>
-
 <div class="row content-task">
+  <div class="row col-md-12 row-text">
```
Comment (0bman) [inline app/views/tasks/index.html.erb]: single quotes

File: app/views/tasks/create.js.erb (hunk 21 above)
Comment (BlackAnge) [reply app/views/tasks/create.js.erb]: no

**Pull request merged**